test/sqllogictest/ldbc_bi.slt

```diff
@@ -0,0 +1,4022 @@
+      count(*) AS popularityScore
+  FROM Message message2
+  JOIN Person_likes_Message like2
+      ON like2.MessageId = message2.MessageId
+  GROUP BY message2.CreatorPersonId;
+
+statement ok
+CREATE INDEX PopularityScoreQ06_person2id ON PopularityScoreQ06 (person2id);
+
+query T multiline
+EXPLAIN WITH poster_w_liker AS (
+        SELECT DISTINCT
+            message1.CreatorPersonId AS person1id,
+            like2.PersonId AS person2id
+        FROM Tag
+        JOIN Message_hasTag_Tag
+          ON Message_hasTag_Tag.TagId = Tag.id
+        JOIN Message message1
+          ON message1.MessageId = Message_hasTag_Tag.MessageId
+        LEFT JOIN Person_likes_Message like2
+               ON like2.MessageId = message1.MessageId
+           -- we don't need the Person itself as its ID is in the like
+         WHERE Tag.name = 'Bob_Geldof'
+    )
+SELECT pl.person1id AS "person1.id",
+       sum(coalesce(ps.popularityScore, 0)) AS authorityScore
+FROM poster_w_liker pl
+LEFT JOIN PopularityScoreQ06 ps
+         ON ps.person2id = pl.person2id
+GROUP BY pl.person1id
+ORDER BY authorityScore DESC, pl.person1id ASC
+LIMIT 100
+----
+Explained Query:
+  Finish order_by=[#1 desc nulls_first, #0 asc nulls_last] limit=100 output=[#0, #1]
+    Return
+      Reduce group_by=[#0] aggregates=[sum(coalesce(#1, 0))]
+        Union
+          Map (null)
+            Union
+              Negate
+                Project (#0)
+                  Get l4
+              Project (#0)
+                Get l3
+          Get l4
+    With
+      cte l4 =
+        Project (#0, #3)
+          Join on=(#1 = #2) type=differential
+            ArrangeBy keys=[[#1]]
+              Filter (#1) IS NOT NULL
+                Get l3
+            ArrangeBy keys=[[#0]]
+              Get materialize.public.popularityscoreq06
+      cte l3 =
+        Distinct group_by=[#0, #1]
+          Union
+            Map (null)
+              Union
+                Negate
+                  Project (#1)
+                    Join on=(#0 = #2) type=differential
+                      ArrangeBy keys=[[#0]]
+                        Project (#1, #2)
+                          Get l2
+                      ArrangeBy keys=[[#0]]
+                        Distinct group_by=[#0]
+                          Project (#1)
+                            Get l1
+                Project (#2)
+                  Get l2
+            Project (#2, #3)
+              Filter (#0 = "Bob_Geldof")
+                Get l1
+      cte l2 =
+        Filter (#0 = "Bob_Geldof")
+          Get l0
+      cte l1 =
+        Project (#0..=#3)
+          Join on=(#1 = #4) type=differential
+            ArrangeBy keys=[[#1]]
+              Get l0
+            ArrangeBy keys=[[#1]]
+              Project (#1, #2)
+                Get materialize.public.person_likes_message
+      cte l0 =
+        Project (#1, #4, #15)
+          Filter (#0) IS NOT NULL
+            Join on=(#0 = #5 AND #4 = #7) type=differential
```

Comment:
I'm sad that the `Filter (#0 = "Bob_Geldof")` got stuck before reaching this Join. When doing measurements with this query, a manual rewrite that would greatly speed this up is to manually push the `WHERE Tag.name = 'Bob_Geldof'` into a subquery in the FROM clause. I.e., instead of `FROM Tag`, you could have `FROM (SELECT * from Tag WHERE Tag.name = 'Bob_Geldof')`. But in the slt EXPLAIN it also makes sense to have this original version.

Btw. this issue reminds me of those subquery issues where cross joins appear due to not being able to push a predicate down enough.

Reply:
This whole CTE is weird: in CTE `l3` we get an extra filter on `#0 = 'Bob_Geldof'`, I think to resolve issues with projected columns. The only use of a raw `l1` is in a join where we'll end up only seeing rows that match anyway (I think).